charms/kfp-ui/tests/unit/test_operator.py

```diff
@@ -297,6 +298,70 @@ def test_install_with_all_inputs(harness, oci_resource_data):
     assert harness.charm.model.unit.status == ActiveStatus()
 
 
+def test_object_storage_secrets_encoded(harness, oci_resource_data):
+    harness.set_leader()
+    http_port = "1234"
+    model_name = "test_model"
+    harness.set_model_name(model_name)
+    harness.update_config({"http-port": http_port})
+
+    kfpui_relation_name = "kfp-ui"
+
+    harness.set_leader()
+
+    # Set up required relations
+    # Future: convert these to fixtures and share with the tests above
+    harness.add_oci_resource(**oci_resource_data)
+
+    # object storage relation
+    os_data = {
+        "_supported_versions": "- v1",
+        "data": yaml.dump(
+            {
+                "access-key": "access-key",
+                "namespace": "namespace",
+                "port": 1234,
+                "secret-key": "secret-key",
```

Comment:
really nitpicky, but I always worry in tests about setting keys and values to the same thing (eg: `{"access-key": "access-key"}`), it can accidentally hide bugs where you use the wrong one by mistake (eg: you meant to use `dict.values` but instead used `dict.keys`)

Reply:
(also, the irony is not lost on me that I bet you followed the pattern from the other tests, and `git blame` points to me for that mistake :) )